File: app/src/main/java/brmobi/moop/features/condominio/CondominioViewHolder.java
package brmobi.moop.features.condominio;

import android.content.Context;
import android.graphics.Color;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import brmobi.moop.model.entities.Condominio;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import brmobi.moop.R;

/**
 * Created by murilo aires on 17/08/2017.
 */

public class CondominioViewHolder extends RecyclerView.ViewHolder {

    private CondominiosFragment fragment;
    private Context context;

    @BindView(R.id.textCondominioNome)
    TextView textNomeCondominio;

    @BindView(R.id.selectable_view)
    View selectableView;

    @OnClick(R.id.selectable_view)
    public void select(View view) {
       fragment.select(getAdapterPosition());
    }

    public CondominioViewHolder(View itemView, CondominiosFragment fragment) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        this.context = itemView.getContext();
        this.fragment = fragment;
    }

    public void bindView(Condominio condominio) {
        textNomeCondominio.setText(condominio.getNome());
        selectableView.setBackgroundColor(condominio.isSelected() ? context.getResources().getColor(R.color.selected_condominio) : Color.WHITE);
    }
}

File: app/src/main/java/brmobi/moop/ui/condominium/add/CondominioViewHolder.kt
package brmobi.moop.ui.condominium.add

import android.content.Context
import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.view.View
import brmobi.moop.R
import brmobi.moop.data.db.model.Condominio
import kotlinx.android.synthetic.main.item_condominio.view.*

/**
 * Created by murilo aires on 17/08/2017.
 */

class CondominioViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val context: Context = itemView.context

    private val textCondominiumName = itemView.textCondominioNome
    val rootView = itemView.selectable_view

    fun bindView(condominio: Condominio) {
        textCondominiumName.text = condominio.nome
        rootView!!.setBackgroundColor(if (condominio.isSelected) context.resources.getColor(R.color.selected_condominio) else Color.WHITE)
    }
}
